cmd/maya-exporter/app/command/exporter_test.go

```diff
@@ -41,39 +40,44 @@ func TestInitialize(t *testing.T) {
 	}
 }
 func TestStartMayaExporter(t *testing.T) {
-	ErrorMessage := make(chan error)
 	cases := map[string]struct {
 		cmdOptions *VolumeExporterOptions
 		err        error
+		targetURL  string
 	}{
-		"If port is busy and path is `/metrics`": {
-			cmdOptions: &VolumeExporterOptions{
-				ControllerAddress: "localhost:9501",
-				MetricsPath:       "/metrics",
-				ListenAddress:     ":9500",
-			},
-			err: errors.New("listen tcp :9500: bind: address already in use"),
+		"Check for metrics": {
+			err:       nil,
+			targetURL: "http://localhost:9500/metrics",
+		},
+		"Check for json": {
+			err:       nil,
+			targetURL: "http://localhost:9500/metrics/?type=json",
 		},
 	}
-	for name, tt := range cases {
-		t.Run(name, func(t *testing.T) {
-			startTestServer(t, tt.cmdOptions, ErrorMessage)
-			msg := <-ErrorMessage
-			if !reflect.DeepEqual(msg.Error(), tt.err.Error()) {
-				t.Fatalf("StartMayaExporter() : expected %v, got %v", tt.err, msg)
-			}
-		})
+
+	options := &VolumeExporterOptions{
+		ControllerAddress: "localhost:9501",
+		MetricsPath:       "/metrics",
+		ListenAddress:     ":9500",
 	}
-}
 
-func startTestServer(t *testing.T, options *VolumeExporterOptions, errMsg chan error) {
 	go func() {
-		//Block port 9500 and attempt to start http server at 9500.
-		listener, err := net.Listen("tcp", "localhost:9500")
-		defer listener.Close()
+		err := options.StartMayaExporter()
```

Comment:
Why are we removing the negative test cases ?
In the earlier code we were testing a scenario where the server was failed to start due to the port 9500 already occupied by some other service. (Only negative test cases are possible)
To verify the above logic (+ve test cases) you need to write integration test.

Reply:
I removed that because I was trying to add UT for endpoints but it was getting very complex as StartMayaExporter is blocking call. and we cannot restart the MayaExporter as the handles will conflict.

Reply:
Yes, that's why a negative test case was written. To test your handler you can move it into another function.